### PR TITLE
fix(note-editor): Switch Image Occlusion types 

### LINKED ISSUE #15579: [Bug] Switching between `Image Occlusion` note types on Add screen causes a crash
###### Reproduction Steps

1. Open App - three dots icon - `Manage note types` - `+` - `Type` - `Add: Image Occlusion` - `OK`
5.  Go back to Deck picker screen
6. `+`  - `Add`
7. `Type:` - change from the already selected `Image Occlusion-xxxxx` to the original) `Image Occlusion`, or vice versa
   - In case that the already selected note type isn't neither, select either one - do the step 2 - 3 - 4  

###### Expected Result
No crash

###### Actual Result
Crash

https://github.com/ankidroid/Anki-Android/assets/10436072/d53be7f2-4081-4d5a-8a66-a3e3375f07ed

###### Debug info
AnkiDroid Version = 2.17beta5 (a06d445bfcea3139accc30faa1b24c0fe214d2b0)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 11 (SDK 30)

ProductFlavor = full

Manufacturer = OPPO

Model = CPH2013

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; CPH2013 Build/RKQ1.200903.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/121.0.6167.165 Mobile Safari/537.36

ACRA UUID = df169a4d-29ec-46b6-b114-141ca7e01ba3

Crash Reports Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Fixes
* Fixes #15579

## Approach
check for null

## How Has This Been Tested?
Unit test

## Learning (optional, can help others)
image

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
For your meme, don't forget "also also Android devs: lateinit"

**Review by mikehardy — APPROVED** [review]
LGTM - only reservation is test code appears to create a new second IO note type for testing even if 2 IO note types already exist. Approving because the fix looks good but maybe a tweak in the test code

File: AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt [hunk 1]
```diff
@@ -170,6 +171,16 @@ interface TestClass {
...
     }
 
+    fun NotetypeJson.createClone(): NotetypeJson {
+        val targetNotetype = requireNotNull(col.notetypes.byName(name)) { "could not find note type '$name'" }
+        val newNotetype = targetNotetype.deepClone().apply {
+            id = 0
+            set("name", "$name+")
+        }
+        col.notetypes.add(newNotetype)
+        return col.notetypes.byName("$name+")!!
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt:181]: Some time ago, I modified `androidTest` to not use the user's collection (I thought they'd appreciate it)

And this PR came after the issue

File: AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.kt [hunk 2]
```diff
@@ -356,12 +357,30 @@ class NoteEditorTest : RobolectricTest() {
         }
     }
 
+    @Ignore("15579")
+    @Test
+    fun `can switch two image occlusion note types 15579`() {
+        val occlusion2 = col.notetypes.byName("Image Occlusion")!!.createClone()
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.kt:362]: Sure!

File: AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt:181]: 🤔  very strange then. You must admit it seems pretty coincidental :-). If you're sure that's not it, then that's not it tho

**Review by mikehardy — APPROVED** [review]
LGTM

File: AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (snowtimeglass) [reply AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt:181]: >why your test collection has a bunch of Image Occlusion, Image Occlusion+, Image Occlusion++ names

Does this seem irrelevant to the several revisions (updates) of Image Occlusion note type in Anki Desktop?

https://github.com/ankitects/anki/releases
Related terms: `Image Occlusion`, `I/O`

[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt:181]: Unsure

[the remark this answers is not in the document]
Comment (dae) [reply AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt:181]: Perhaps it's adding a new notetype each time you open the add screen? It's delegating to the backend addImageOcclusionNotetype(), which should reuse an existing notetype provided originalStockKind is correctly set, and it has 4+ fields.

**Review by lukstbit — APPROVED** [review]

**Pull request merged**